### ISSUE TITLE
SNOW-152592: Support for setting 'APPLICATION' parameter

### ISSUE THREAD

**Issue by BenBirt** [snowflakedb/snowflake-connector-nodejs#100]
The JDBC client equivalent is here: https://github.com/snowflakedb/snowflake-jdbc/blob/d045379ea345a5beadacc279a28269cf0c2047ef/src/main/java/net/snowflake/client/core/SessionUtil.java#L461

Right now I have a (very ugly) workaround (I'm not 100% sure it works yet), it would be good to have first-class API support for this:

```
require("snowflake-sdk/lib/core")({
  httpClientClass: require("snowflake-sdk/lib/http/node"),
  loggerClass: require("snowflake-sdk/lib/logger/node"),
  client: {
    version: require("snowflake-sdk/lib/util").driverVersion,
    environment: {
      ...process.versions,
      APPLICATION: "<App name>"
    }
  }
})
```

**Comment by sfc-gh-hkapre** [comment]
This is in plan. Targeting having this by the end of October.

**Comment by sfc-gh-dszmolka** [comment]
support for setting 'Application' in connection released with v1.6.9, thank you for your patience
